modeladminutils/edit_handlers.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import absolute_import, unicode_literals
+
+from django.template.loader import render_to_string
+from django.utils.safestring import mark_safe
+
+from wagtail.wagtailadmin.edit_handlers import BaseChooserPanel
+
+from modeladminutils.widgets import GenericModelChooser
+
+
+class BaseGenericModelChooserPanel(BaseChooserPanel):
```

Comment:
i doubt we can user snippets panel without actually using snippets

snippets is an easier approach
i think i should have used it instead of implementing this panel

but now when my widget is implemented, modeladmin approach opens more opportunities